### ISSUE TITLE
import-order request: order by default import name

### ISSUE THREAD

**Issue by dawsonc623** [palantir/tslint#3994]
At my workplace we use default exports extensively and prefer to order my imports by the name we assign to the default exports. Example:

```ts
// Incorrect
import foo from "bar";
import baz from "biz";

// Correct
import baz from "biz";
import foo from "bar";
```

Currently, there is no way to enforce this (as far as I can see). I believe an addition to the options given for `ordered-imports` would be useful for handling this case.

**Comment by giladgray** [comment]
🤔 i find ordering by paths to be the most reliable order because it's deterministic, whereas your suggestion above is not (you can name default exports whatever you want).

i am not keen on this suggestions, but if you submit a PR then i will review it 👍

**Comment by giladgray** [comment]
tech-wise, this should be an option on `ordered-imports` as you suggested.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
